waku/v2/protocol/rln/group_manager/root_tracker.go

```diff
@@ -102,18 +101,18 @@ func (m *MerkleRootTracker) IndexOf(root [32]byte) int {
 
 // UpdateLatestRoot should be called when a block containing a new
 // IDCommitment is received so we can keep track of the merkle root change
-func (m *MerkleRootTracker) UpdateLatestRoot(blockNumber uint64) (rln.MerkleNode, error) {
+func (m *MerkleRootTracker) UpdateLatestRoot(blockNumber uint64) rln.MerkleNode {
 	m.Lock()
 	defer m.Unlock()
 
 	root, err := m.rln.GetMerkleRoot()
 	if err != nil {
-		return [32]byte{}, err
+		utils.Logger().Named("root-tracker").Panic("could not retrieve merkle root", zap.Error(err))
```

Comment:
What would be the recovery here? 
If it just restarting the node, I am wondering if we should retry once more before declaring panic.

Reply:
The recovery is a restart. I'm not sure if we should retry. I'm worried about RLN being in some weird state after running into this error, and a restart wouldn't hurt much the sync process since it would still start from the last record synced instead of from scratch